odl/discr/lp_discr.py

```diff
@@ -286,12 +286,24 @@ def element(self, inp=None, **kwargs):
         else:
             # Sequence-type input
             arr = np.asarray(inp, dtype=self.dtype, order=self.order)
-            if arr.ndim > 1 and arr.shape != self.shape:
+            input_shape = arr.shape
+
+            # Attempt to handle some cases with exact match
+            if arr.ndim > 1 and self.ndim == 1:
                 arr = np.squeeze(arr)  # Squeeze could solve the problem
                 if arr.shape != self.shape:
                     raise ValueError(
-                        'input shape {} does not match grid shape {}.'
-                        ''.format(arr.shape, self.shape))
+                        'input shape {} does not match grid shape {}'
+                        ''.format(input_shape, self.shape))
+            elif arr.shape != self.shape and arr.ndim == self.ndim:
+                # Try to broadcast the array if possible
+                for i in range(self.ndim):
+                    if arr.shape[i] == 1 and self.shape[i] != 1:
+                        arr = arr.repeat(self.shape[i], axis=i)
```

Comment:
Did you benchmark this? I think this solution is quite slow compared to broadcasting with factor zero. I suggest that you test this and try it against the alternative of building a zero array for the missing dimensions, i.e.
```python
shape = [m if n == 1 and m != 1 else n
         for n, m in zip(arr.shape, self.shape)]
arr = arr + np.zeros(shape)
```